calculation/gmhazard_calc/gmhazard_calc/directivity/directivity.py

```diff
@@ -157,40 +159,63 @@ def compute_fault_directivity(
         constants.EventType.from_rake(rake),
     )
 
-    # Creating the array to store all fdi values
-    fdi_array = np.zeros((n_hypo_data.nhypo, len(sites), len(periods)))
-    phired_array = np.zeros((n_hypo_data.nhypo, len(sites), len(periods)))
-    for index, planes in enumerate(hypo_planes):
-        # Gets the plane index of the hypocentre
-        plane_index = plane_ind[index]
+    if n_procs == 1:
+        fdi, phired = [], []
+        for ix, cur_planes in enumerate(hypo_planes):
+            cur_fdi, (cur_phi_red, cur_predictor_functions, cur_other) = _compute_directivity_effect(
+                lon_lat_depth,
+                cur_planes,
+                plane_ind[ix],
+                sites,
+                nominal_strike,
+                nominal_strike2,
+                mag,
+                rake,
+                periods,
+            )
 
-        fdi, (phi_red, predictor_functions, other) = _compute_directivity_effect(
-            lon_lat_depth,
-            planes,
-            plane_index,
-            sites,
-            nominal_strike,
-            nominal_strike2,
-            mag,
-            rake,
-            periods,
-        )
+            fdi.append(cur_fdi)
+            phired.append(cur_phi_red)
+    else:
+        # Compute directivity for all hypocentres
+        with mp.Pool(n_procs) as pool:
+            results = pool.starmap(
+                _compute_directivity_effect,
+                [
+                    (
+                        lon_lat_depth,
+                        cur_planes,
+                        plane_ind[ix],
+                        sites,
+                        nominal_strike,
+                        nominal_strike2,
+                        mag,
+                        rake,
+                        periods,
+                    )
+                    for (ix, cur_planes) in enumerate(hypo_planes)
+                ],
+            )
+
+        # Combine results
+        fdi = [cur_result[0] for cur_result in results]
+        phired = [cur_result[1][0] for cur_result in results]
 
-        fdi_array[index] = fdi
-        phired_array[index] = phi_red
+    fdi = np.stack(fdi, axis=0)
+    phired = np.stack(phired, axis=0)
```

Comment:
I'd prefer this with the stack done on the one line - I think python can optimise the nested call a little.

Reply:
I'm not  100% sure what you mean. Do you mean making it a one-liner like this `fdi, phired = np.stack(fdi, axis=0), np.stack(phired, axis=0)`?

Reply:
I mean like `fdi = np.stack([cur_result[0] for cur_result in results], axis=0)`

or convert the whole results "mess" into a numpy array and slice it.

Reply:
Then I would have to indent it and do a separate stack for the for loop. 
And I don't think you can just turn into a numpy array and slice it, since its a Tuple that contains another tuple.

Reply:
Oh yep. I didn't notice the indentation.